src/foam/nanos/auth/Group.js

```diff
@@ -211,39 +211,57 @@ foam.CLASS({
         }
       ],
       javaCode: `
-AppConfig config = (AppConfig) ((AppConfig) x.get("appConfig")).fclone();
+        // Find Group details, by iterating up through group.parent
+        AppConfig config          = (AppConfig) ((AppConfig) x.get("appConfig")).fclone();
+        String configUrl          = "";
+        String configSupportEmail = "";
+        Boolean urlFound          = false;
+        Boolean supportEmailFound = false;
+        Group group               = this;
+        String grp                = "";
+        DAO groupDAO              = (DAO) x.get("groupDAO");
 
-String configUrl = config.getUrl();
+        while ( group != null && ! (urlFound && supportEmailFound)) {
+          configUrl          = urlFound ? configUrl : group.getUrl();
+          configSupportEmail = supportEmailFound ? configSupportEmail : group.getSupportEmail();
+      
+          // Once true, stay true
+          urlFound          = urlFound   ? urlFound   : ! SafetyUtil.isEmpty(configUrl);
+          supportEmailFound = supportEmailFound ? supportEmailFound : ! SafetyUtil.isEmpty(configSupportEmail);
```

Comment:
Little optimization - to break when both have been found.  Otherwise looks good.

Reply:
good catch thx 👍